pkg/networkmanager/network-main.jsx

```diff
@@ -0,0 +1,283 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import $ from 'jquery';
+import cockpit from "cockpit";
+import React from 'react';
+
+import {
+    Button,
+    Card, CardActions, CardBody, CardTitle, CardHeader,
+    Gallery,
+    Page, PageSection, PageSectionVariants,
+    Text, TextVariants,
+} from "@patternfly/react-core";
+import { cellWidth } from '@patternfly/react-table';
+
+import { FirewallSwitch } from "./firewall-switch.jsx";
+import { ListingTable } from "cockpit-components-table.jsx";
+import { LogsPanel } from "cockpit-components-logs-panel.jsx";
+import { NetworkPageDialogs } from './network-main-dialogs.jsx';
+import { NetworkPlots } from "./plots";
+
+import firewall from './firewall-client.js';
+import {
+    device_state_text,
+    is_managed,
+    render_active_connection,
+} from './interfaces.js';
+import { PlotState } from 'plot';
+import { getNetworkGraphMetrics } from './helpers.js';
+
+import "plot.css";
+
+const _ = cockpit.gettext;
+
+export class NetworkPage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            firewall,
+            managed: [],
+            unmanaged: [],
+            usageSamples: { },
+            usage_metrics_channel: getNetworkGraphMetrics(),
```

Comment:
There should be only one global usage metrics channel.  Likewise, the two PlotState objects (one here and one in NetworkInterfacePage) should be global.
Now new ones are constructed everytime the user switches between NetworkPage and NetworkInterfacePage, and the old ones will never be disposed off.  This probably means we are opening more and more channels to the bridge.

Reply:
Global means inside the App instance.  So I guess we should add them to the ModelContext somehow?

Reply:
I have added a FIXUP for this.